src/main/java/com/cloudbees/jenkins/support/configfiles/AgentsConfigFile.java

```diff
@@ -0,0 +1,66 @@
+package com.cloudbees.jenkins.support.configfiles;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.util.Helper;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.transform.TransformerException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Adds nodes config.xml files to the bundle.
+ * Any secret string is redacted in the xml file.
+ *
+ * Unselected by default.
+ */
+@Extension
+public class AgentsConfigFile extends Component {
+
+    @Override
+    public boolean isSelectedByDefault() {
+        return false;
+    }
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return Messages.AgentsConfigFile_displayName();
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        File[] agentDirs = new File(Helper.getActiveInstance().getRootDir(), "nodes").listFiles();
+        if (agentDirs == null) {
+            return;
+        }
+        for(File agentDir : agentDirs) {
+            File config = new File(agentDir, "config.xml");
+            try {
+                File patchedConfig = SecretHandler.findSecrets(config);
+                container.add(new TemporalFileContent("nodes/slave/" + agentDir.getName() + "/config.xml", patchedConfig));
+            } catch (IOException | ParserConfigurationException | XMLStreamException | SAXException | TransformerException e) {
+                LOGGER.log(Level.WARNING, "could not add the {0} configuration file to the support bundle because of: {1}", new Object[]{ config.getAbsolutePath(), e });
```

Comment:
Not sure if passing the exception in that string is gonna be beautiful. Maybe:

```java
LogRecord record = new LogRecord(Level.WARNING, "Could not add the {0} configuration file to the support bundle because of: {1}");
record.setParameters(new Object[]{config.getAbsolutePath(), e});
record.setThrown(e);
LOGGER.log(record);
```